src/cas/__main__.py

```diff
@@ -43,12 +43,11 @@ def main():
         json_file_path = args.json
         anndata_file_path = args.anndata
         output_file_path = args.output
-        validate = args.validate
 
-        if anndata_file_path == output_file_path:
+        if output_file_path and anndata_file_path == output_file_path:
```

Comment:
We can use absolute paths for path comparison in case one is a relative path and other is absolute and we can't detect that they are same. 
```
if output_file_path and os.path.abspath(anndata_file_path) ==os.path.abspath(output_file_path):
```